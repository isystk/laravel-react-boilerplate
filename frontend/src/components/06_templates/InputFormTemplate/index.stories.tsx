import { Meta, Story } from '@storybook/react'
import React from 'react'
import InputFormTemplate, { InputFormTemplateProps } from './index'
import MainService from '@/services/main'

export default {
  title: '06_templates/InputFormTemplate',
  component: InputFormTemplate,
} as Meta

const Template: Story = () => {
  const props: InputFormTemplateProps = { title: 'サンプル' }
  return <InputFormTemplate {...props} />
}

export const Primary = Template.bind({})
Primary.storyName = 'プライマリ'
Primary.args = {}
