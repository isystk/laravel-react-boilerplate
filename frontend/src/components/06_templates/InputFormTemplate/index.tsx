import Circles from '@/components/02_interactions/Circles'
import React, { FC, useState } from 'react'
import { ContainerProps, WithChildren } from 'types'
import { connect } from '@/components/hoc'
import HtmlSkeleton, {
  NoIndex,
  Title,
} from '@/components/05_layouts/HtmlSkeleton'
import Header from '@/components/04_organisms/Header'
import SlideMenu from '@/components/04_organisms/SlideMenu'
import { frontMenuItems } from '@/constants/menu'
import MainService from '@/services/main'

/** InputFormTemplateProps Props */
export type InputFormTemplateProps = WithChildren & {
  main: MainService
  title: string
}
/** Presenter Props */
export type PresenterProps = InputFormTemplateProps

/** Presenter Component */
const InputFormTemplatePresenter: FC<PresenterProps> = ({
  children,
  title,
  isMenuOpen,
  setMenuOpen,
  ...props
}) => (
  <HtmlSkeleton>
    <Title>{title}</Title>
    <NoIndex />
    <Header
      isMenuOpen={isMenuOpen}
      setMenuOpen={setMenuOpen}
      menuItems={frontMenuItems}
    />
    <div className="flex items-center justify-center">
      <Circles>
        <div className="px-1 md:px-32 w-full">
          <div className="py-8">{children}</div>
        </div>
      </Circles>
    </div>
    <div className="md:hidden">
      <SlideMenu
        isMenuOpen={isMenuOpen}
        setMenuOpen={setMenuOpen}
        menuItems={frontMenuItems}
      />
    </div>
  </HtmlSkeleton>
)

/** Container Component */
const InputFormTemplateContainer: React.FC<
  ContainerProps<InputFormTemplateProps, PresenterProps>
> = ({ presenter, children, ...props }) => {
  const [isMenuOpen, setMenuOpen] = useState(false)
  return presenter({
    children,
    isMenuOpen,
    setMenuOpen,
    ...props,
  })
}

export default connect<InputFormTemplateProps, PresenterProps>(
  'InputFormTemplate',
  InputFormTemplatePresenter,
  InputFormTemplateContainer
)
