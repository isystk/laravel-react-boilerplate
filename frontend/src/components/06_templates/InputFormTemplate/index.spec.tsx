import React from 'react'
import { render } from '@testing-library/react'
import InputFormTemplate, { InputFormTemplateProps } from './index'

describe('LandingPageTemplate', () => {
  it('Match Snapshot', () => {
    const props: InputFormTemplateProps = { title: 'サンプル' }
    const { asFragment } = render(<InputFormTemplate {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
