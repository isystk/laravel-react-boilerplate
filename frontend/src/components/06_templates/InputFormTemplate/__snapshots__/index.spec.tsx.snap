// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPageTemplate Match Snapshot 1`] = `
<DocumentFragment>
  <header
    class="css-0 bg-main z-50 flex justify-between items-center h-16 px-4 py-4 sm:px-8 w-full"
  >
    <a
      class="css-0"
      href="#"
    >
      <div
        class="w-20 md:w-40"
      >
        <img
          class="css-s08p0c "
          src="/images/logo.png"
        />
      </div>
    </a>
    <div
      class="flex pr-3 z-50 md:hidden "
    >
      <button
        class="space-y-2 css-jgxxba"
      >
        <div
          class="w-6 h-0.5 bg-black"
        />
        <div
          class="w-6 h-0.5 bg-black"
        />
        <div
          class="w-6 h-0.5 bg-black"
        />
      </button>
    </div>
    <nav
      class="hidden sm:block"
    >
      <ul
        class="flex gap-6"
      >
        <li>
          <a
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li>
          <a
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <div
    class="flex items-center justify-center"
  >
    <div
      class="css-1wwfi4"
    >
      <ul
        class="css-10ffb9g"
      >
        <li
          class="css-i6ammj"
        />
        <li
          class="css-l07tar"
        />
        <li
          class="css-1hre2z7"
        />
        <li
          class="css-p2xyi8"
        />
        <li
          class="css-u8edpl"
        />
        <li
          class="css-ik3prs"
        />
        <li
          class="css-1wh69e5"
        />
        <li
          class="css-1cg3at6"
        />
        <li
          class="css-1iaru34"
        />
        <li
          class="css-1io9drp"
        />
      </ul>
      <div
        class="relative"
      >
        <div
          class="px-1 md:px-32 w-full"
        >
          <div
            class="py-8"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="md:hidden"
  >
    <div
      class="undefined fixed top-0 right-0 z-30 bg-base h-screen w-64 css-hc9ocb"
    >
      <div
        class="mt-24"
      />
      <ul
        class="list-none"
      >
        <li
          class="py-4"
        >
          <a
            class="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/"
            rel=""
          >
            Home
          </a>
        </li>
        <li
          class="py-4"
        >
          <a
            class="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="http://localhost:9000/product/prod_NpvV9ohJIlgElI/payment"
            rel=""
          >
            Rates
          </a>
        </li>
        <li
          class="py-4"
        >
          <a
            class="block text-gray-700 font-bold whitespace-nowrap px-4"
            href="https://blog.isystk.com/contact/"
            rel="noreferrer"
            target="_blank"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </div>
    <div
      class="hidden z-20 bg-black opacity-50 fixed top-0 w-full h-screen"
    />
  </div>
</DocumentFragment>
`;
